tests/bluetooth/bsim_bt/bsim_test_l2cap_ecred/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,503 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright (c) 2021 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+
+#include <zephyr/types.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/l2cap.h>
+#include "bs_types.h"
+#include "bs_tracing.h"
+#include "bstests.h"
+
+#define LOG_MODULE_NAME main_l2cap_ecred
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+extern enum bst_result_t bst_result;
+
+#define FAIL(...)				       \
+	do {					       \
+		bst_result = Failed;		       \
+		bs_trace_error_time_line(__VA_ARGS__); \
+	} while (0)
+
+#define PASS(...)				    \
+	do {					    \
+		bst_result = Passed;		    \
+		bs_trace_info_time(1, __VA_ARGS__); \
+	} while (0)
+
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+};
+
+#define DATA_MTU 2000
+#define DATA_MPS 65
+#define DATA_BUF_SIZE BT_L2CAP_SDU_BUF_SIZE(DATA_MTU)
+#define L2CAP_CHANNELS 5
+#define SERVERS 1
+
+NET_BUF_POOL_FIXED_DEFINE(rx_data_pool, L2CAP_CHANNELS, BT_L2CAP_BUF_SIZE(DATA_BUF_SIZE), 8, NULL);
+
+NET_BUF_POOL_FIXED_DEFINE(tx_data_pool0, 1, BT_L2CAP_BUF_SIZE(DATA_MTU), 0, NULL);
+
+NET_BUF_POOL_FIXED_DEFINE(tx_data_pool1, 1, BT_L2CAP_BUF_SIZE(DATA_MTU), 0, NULL);
+
+static struct channel {
+	uint8_t chan_id; /* Internal number that identifies L2CAP channel. */
+	struct bt_l2cap_le_chan le;
+	bool in_use;
+	uint32_t time_of_first_received_sdu;
+} channels[L2CAP_CHANNELS];
+static bool volatile is_connected;
+static struct bt_l2cap_server servers[SERVERS];
+
+static struct channel *get_free_channel(void)
+{
+	uint8_t i;
+	struct channel *chan;
+
+	for (i = 0U; i < L2CAP_CHANNELS; i++) {
+		if (channels[i].in_use) {
+			continue;
+		}
+		chan = &channels[i];
+		(void)memset(chan, 0, sizeof(*chan));
+		chan->chan_id = i;
+		channels[i].in_use = true;
+		return chan;
+	}
+
+	return NULL;
+}
+
+static struct net_buf *chan_alloc_buf_cb(struct bt_l2cap_chan *chan)
+{
+	LOG_DBG("Allocated on chan %p", chan);
+	return net_buf_alloc(&rx_data_pool, K_FOREVER);
+}
```

Comment:
Looks like the `chan` pointer is only used for logging purposes. Is that intentional? 
```suggestion
static struct net_buf *chan_alloc_buf_cb(struct bt_l2cap_chan *chan)
{
	LOG_DBG("Allocated on chan %p", chan);
	return net_buf_alloc(&rx_data_pool, K_FOREVER);
}
```